File: src/SOP8/Base/History/Base.History.IBLL/Response/ResponseSOPHistory.cs
using System.Collections.Generic;
using Response;

namespace Base.History.IBLL.Response
{
    using Models.SOP;

    public class ResponseSOPHistory : MessageResult
    {
        private List<SOPHistoryData> m_sopHistoryDatas = new List<SOPHistoryData>();
        private int m_nTotalCount = -1;

        public List<SOPHistoryData> SOPHistoryDatas
        {
            get { return m_sopHistoryDatas; }
            set { m_sopHistoryDatas = value; }
        }

        public int TotalCount
        {
            get { return m_nTotalCount; }
            set { m_nTotalCount = value; }
        }

        public ResponseSOPHistory()
            : base()
        {
        }

        public ResponseSOPHistory(bool success, string message)
            : base(success, message)
        {
        }
    }
}

File: src/SOP8/Egovframework/Base/base-history-ibll/src/main/java/egovframework/base/history/ibll/response/ResponseSOPHistory.java
package egovframework.base.history.ibll.response;

import egovframework.base.history.ibll.models.sop.SOPHistoryData;
import egovframework.base.response.MessageResult;

import java.util.ArrayList;
import java.util.List;

public class ResponseSOPHistory extends MessageResult {

    private List<SOPHistoryData> sopHistoryDatas = new ArrayList<>();
    private int totalCount = -1;

    // Constructors
    public ResponseSOPHistory() {
        super();
    }

    public ResponseSOPHistory(boolean success, String message) {
        super(success, message);
    }

    // Getters and Setters
    public List<SOPHistoryData> getSopHistoryDatas() {
        return sopHistoryDatas;
    }

    public void setSopHistoryDatas(List<SOPHistoryData> sopHistoryDatas) {
        this.sopHistoryDatas = sopHistoryDatas;
    }

    public int getTotalCount() {
        return totalCount;
    }

    public void setTotalCount(int totalCount) {
        this.totalCount = totalCount;
    }
}